examples/add/train_add_distill_sd_wds.py

```diff
@@ -0,0 +1,1673 @@
+    )
+
+    # Make one log on every process with the configuration for debugging.
+    logging.basicConfig(
+        format="%(asctime)s - %(levelname)s - %(name)s - %(message)s",
+        datefmt="%m/%d/%Y %H:%M:%S",
+        level=logging.INFO,
+    )
+    logger.info(accelerator.state, main_process_only=False)
+    if accelerator.is_local_main_process:
+        transformers.utils.logging.set_verbosity_warning()
+        diffusers.utils.logging.set_verbosity_info()
+    else:
+        transformers.utils.logging.set_verbosity_error()
+        diffusers.utils.logging.set_verbosity_error()
+
+    # If passed along, set the training seed now.
+    if args.seed is not None:
+        set_seed(args.seed)
+
+    # Handle the repository creation
+    if accelerator.is_main_process:
+        if args.output_dir is not None:
+            os.makedirs(args.output_dir, exist_ok=True)
+
+        if args.push_to_hub:
+            create_repo(
+                repo_id=args.hub_model_id or Path(args.output_dir).name,
+                exist_ok=True,
+                token=args.hub_token,
+                private=True,
+            ).repo_id
+
+    # 1. Create the noise scheduler and the desired noise schedule.
+    # Enforce zero terminal SNR (see section 3.1 of ADD paper)
+    # TODO: is there a better way to implement this?
+    teacher_scheduler = DDIMScheduler.from_pretrained(
+        args.pretrained_teacher_model, subfolder="scheduler", revision=args.teacher_revision
+    )
+    if not teacher_scheduler.config.rescale_betas_zero_snr:
+        teacher_scheduler.config["rescale_betas_zero_snr"] = True
+    noise_scheduler = DDIMScheduler(**teacher_scheduler.config)
+
+    # DDIMScheduler calculates the alpha and sigma noise schedules (based on the alpha bars) for us
+    alpha_schedule = torch.sqrt(noise_scheduler.alphas_cumprod)
+    sigma_schedule = torch.sqrt(1 - noise_scheduler.alphas_cumprod)
+    # denoiser gets predicted original sample x_0 from prediction_type using alpha and sigma noise schedules
+    denoiser = Denoiser(alpha_schedule, sigma_schedule)
+
+    # Create time-dependent weighting schedule c(t) for scaling the GAN generator reconstruction loss term.
+    if args.weight_schedule == "uniform":
+        train_weight_schedule = torch.ones_like(noise_scheduler.alphas_cumprod)
+    elif args.weight_schedule == "exponential":
+        # Set gamma schedule equal to alpha_bar (`alphas_cumprod`) schedule. Higher timesteps have less weight.
+        train_weight_schedule = noise_scheduler.alphas_cumprod
+    elif args.weight_schedule == "sds":
+        # Score distillation sampling weighting
+        # Introduced in the DreamFusion paper: https://arxiv.org/pdf/2209.14988.pdf.
+        raise NotImplementedError("SDS distillation weighting is not yet implemented.")
```

Comment:
I think this is given in apendix A of ADD paper:  it should be `(alpha_t / 2*sigma_t) * w(t)`

Reply:
If my understanding of the [ADD](https://arxiv.org/pdf/2311.17042.pdf) and [DreamFusion](https://arxiv.org/pdf/2209.14988.pdf) (which introduced SDS) papers is correct, for SDS weighting we are free in our choice of $`w(t)`$, right? In particular, we are free to choose $`w(t) = 1`$ so that $`c(t)`$ reduces to $`\frac{\alpha_t}{2\sigma_t}`$.

Reply:
Yeah, right `w(t)` can be chosen by us.

Reply:
we can set the default of 1.